jans-cedarling/bindings/cedarling_python/src/authorize/diagnostics.rs

```diff
@@ -0,0 +1,50 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+use super::policy_evaluation_error::PolicyEvaluationError;
+use cedarling::bindings::cedar_policy;
+use std::collections::HashSet;
+
+use pyo3::prelude::*;
+
+/// Diagnostics
+/// ===========
+///
+/// Provides detailed information about how a policy decision was made, including policies that contributed to the decision and any errors encountered during evaluation.
+///
+/// Attributes
+/// ----------
+/// reason : set of str
+///     A set of `PolicyId`s for the policies that contributed to the decision. If no policies applied, this set is empty.
+/// errors : list of PolicyEvaluationError
+///     A list of errors that occurred during the authorization process. These are unordered as policies may be evaluated in any order.
+#[derive(Debug, Clone)]
+#[pyclass(get_all)]
+pub struct Diagnostics {
```

Comment:
I'm wondering what usage patterns a python developer would need here to make use of the diagnostics in a reasonably ergonomic way.

Reply:
We need to allow users to debug their policies.

Reply:
Agreed. I'm wondering how much thought to put into how easy or hard that is.

Reply:
You can find example of usage in the `example.py` file.

Actually, I tried to make API similar to rust. To be almost the same in different language bindings.